blueflood-core/src/test/java/com/rackspacecloud/blueflood/types/BluefloodCounterRollupTest.java

```diff
@@ -1,93 +1,758 @@
+        //    ------+-----------------------+-----------------------+-----------------------+-----
+        //          1                       1                       1                       1
+        //          2                       2                       2                       2
+        //          3                       3                       3                       3
+        //          4                       5                       6                       7
+        //          L                       L                       L                       L
+        //          |                                                                       |
+        //          | <------------------------- three time units ------------------------> |
+        //                                         (3+4+5)/3=4
+        //
+
+        // given
+        Points<SimpleNumber> input = new Points<SimpleNumber>();
+        input.add(new Points.Point<SimpleNumber>(1234L, new SimpleNumber(3L)));
+        input.add(new Points.Point<SimpleNumber>(1235L, new SimpleNumber(4L)));
+        input.add(new Points.Point<SimpleNumber>(1236L, new SimpleNumber(5L)));
+        // when
+        BluefloodCounterRollup rollup = BluefloodCounterRollup.buildRollupFromRawSamples(input);
+        // then
+        Number count = rollup.getCount();
+        assertNotNull(count);
+        assertThat(count, is(CoreMatchers.<Number>instanceOf(Long.class)));
+        assertEquals(12L, count.longValue());
+        assertEquals(6.0d, rollup.getRate(), 0.00001d); // (3+4+5)/(3-1)=6
+        assertEquals(3, rollup.getSampleCount());
+    }
+
+    @Test
+    public void rawSampleBuilderWithFourInputsYieldsRateEqualToOffByOneAverage() throws IOException {
+
+        // given
+        Points<SimpleNumber> input = new Points<SimpleNumber>();
+        input.add(new Points.Point<SimpleNumber>(1234L, new SimpleNumber(3L)));
+        input.add(new Points.Point<SimpleNumber>(1235L, new SimpleNumber(4L)));
+        input.add(new Points.Point<SimpleNumber>(1236L, new SimpleNumber(5L)));
+        input.add(new Points.Point<SimpleNumber>(1237L, new SimpleNumber(6L)));
+        // when
+        BluefloodCounterRollup rollup = BluefloodCounterRollup.buildRollupFromRawSamples(input);
+        // then
+        Number count = rollup.getCount();
+        assertNotNull(count);
+        assertThat(count, is(CoreMatchers.<Number>instanceOf(Long.class)));
+        assertEquals(18L, count.longValue());
+        assertEquals(6.0d, rollup.getRate(), 0.00001d); // (3+4+5+6)/(4-1)=6
+        assertEquals(4, rollup.getSampleCount());
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void rawSampleBuilderWithNullInputsThrowsException() throws IOException {
+
+        // when
+        BluefloodCounterRollup rollup = BluefloodCounterRollup.buildRollupFromRawSamples(null);
+        // then
+        // the exception is thrown
```

Comment:
Eh, it doesn't bother me so much. My use of given/when/then/expect is a holdover from using spock. The labels made the tests more readable. They are one of my favorite features of the framework.